lib/src/models/movie.dart

```diff
@@ -1,28 +1,32 @@
 import 'movie_result.dart';
 
 class Movie {
-  int _page;
-  int _total_results;
-  int _total_pages;
-  List<MovieResult> _results = [];
+  late int page;
+  late int totalResults;
+  late int totalPages;
+  late List<MovieResult> results;
 
-  Movie.fromJson(Map<String, dynamic> json) {
-    _page = json['page'];
-    _total_results = json['total_results'];
-    _total_pages = json['total_pages'];
-    List<MovieResult> temp = [];
-    for (int i = 0; i < json['results'].length; i++) {
-      MovieResult result = MovieResult(json['results'][i]);
-      temp.add(result);
-    }
-    _results = temp;
-  }
-
-  List<MovieResult> get results => _results;
+  Movie({
+    required this.page,
+    required this.results,
+    required this.totalResults,
+    required this.totalPages,
+  });
 
-  int get total_pages => _total_pages;
+  factory Movie.fromJson(
+    Map<String, dynamic> parsedJson,
+  ) {
+    var parsedJsonList = parsedJson['results'] as List;
 
-  int get total_results => _total_results;
+    List<MovieResult> movieTemporary = parsedJsonList
```

Comment:
Instead of movieTemporary do you think that is more descriptive to change it for movieResultList? Or something like that.

Reply:
done